cmd/submit.go

```diff
@@ -125,7 +125,13 @@ func runSubmit(cfg config.Config, flags *pflag.FlagSet, args []string) error {
 	}
 
 	exercise := workspace.NewExerciseFromDir(exerciseDir)
-
+	migrationStatus, err := exercise.MigrateLegacyMetadataFile()
+	if err != nil {
+		return err
+	}
+	if verbose, _ := flags.GetBool("verbose"); verbose {
+		fmt.Fprintf(os.Stderr, migrationStatus.String(exercise))
```

Comment:
Oh, I see what you mean by it seems wrong that it takes an exercise! Yes, somehow I didn't see that it was the `String()` method.

Yes, I agree that this isn't ideal.